internal/template/funcmap.go

```diff
@@ -156,3 +157,16 @@ func schemaDescription(s string) string {
 
 	return fmt.Sprintf(`"%s"`, s)
 }
+
+// substr returns a substring that starts at index 'start'
+// and spans 'length' characters (or until the end of the string,
+// whichever comes first).
+func substr(s string, start int, length int) string {
+	if start >= len(s) {
```

Comment:
nit: lets return (string, error) with an error message saying what the problem is